### PR TITLE
cilium: install transient rules during agent restart

### PR DESCRIPTION
**Pull request by borkmann**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8908)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
```
16:37:52      k8s1-1.15: Ign:29 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages
16:37:52      k8s1-1.15: Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse i386 Packages [8,852 B]
16:37:52      k8s1-1.15: Err:28 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse i386 Packages
16:37:52      k8s1-1.15:   File has unexpected size (6784 != 8852). Mirror sync in progress? [IP: 91.189.88.24 80]
16:37:52      k8s1-1.15:   Hashes of expected file:
16:37:52      k8s1-1.15:    - Filesize:8852 [weak]
16:37:52      k8s1-1.15:    - SHA256:649e2769a3136d4cca19f5c9fcf9c6ec6de8133dabcfcbaa789bdb6c8c083435
16:37:52      k8s1-1.15:    - SHA1:7194d1d10a8fc88b63e6e1e9af46a276f2304680 [weak]
16:37:52      k8s1-1.15:    - MD5Sum:e770a1e882af0807c63f86ba309959c5 [weak]
16:37:52      k8s1-1.15:   Release file created at: Wed, 14 Aug 2019 03:39:03 +0000
16:37:52      k8s1-1.15: Get:27 http://archive.ubuntu.com/ubuntu bionic-updates/universe Translation-en [419 kB]
16:37:52      k8s1-1.15: Err:27 http://archive.ubuntu.com/ubuntu bionic-updates/universe Translation-en
16:37:52      k8s1-1.15:   
16:37:52      k8s1-1.15: Get:29 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [11.9 kB]
16:37:52      k8s1-1.15: Err:29 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages
16:37:52      k8s1-1.15:   
16:38:01      k8s1-1.15: Fetched 2,237 kB in 34s (65.9 kB/s)
16:38:01      k8s1-1.15: Reading package lists...
16:38:06      k8s1-1.15: E: Failed to stat /var/lib/apt/lists/partial/archive.ubuntu.com_ubuntu_dists_bionic-backports_InRelease - pkgAcqTransactionItem::TransactionState-stat (2: No such file or directory)
16:38:06      k8s1-1.15: E: Failed to stat /var/lib/apt/lists/partial/security.ubuntu.com_ubuntu_dists_bionic-security_InRelease - pkgAcqTransactionItem::TransactionState-stat (2: No such file or directory)
16:38:06      k8s1-1.15: E: Failed to stat /var/lib/apt/lists/partial/security.ubuntu.com_ubuntu_dists_bionic-security_main_binary-amd64_Packages - pkgAcqTransactionItem::TransactionState-stat (2: No such file or directory)
16:38:06      k8s1-1.15: E: Failed to stat /var/lib/apt/lists/partial/security.ubuntu.com_ubuntu_dists_bionic-security_main_binary-i386_Packages - pkgAcqTransactionItem::TransactionState-stat (2: No such file or directory)
```

**Comment by borkmann** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

File: daemon/datapath.go [hunk 1]
```diff
@@ -199,10 +199,17 @@ func (d *Daemon) compileBase() error {
 		return err
 	}
 
+	if option.Config.InstallIptRules {
```
Comment (brb) [inline daemon/datapath.go:202]: We have more places in the code where we do `RemoveRules()` &`InstallRules()`. So perhaps it makes sense to move `TransientRulesStart()` to `RemoveRules()` and introduce the arg `addTransient bool` to `RemoveRules()` (accordingly, `delTransient` to `InstallRules()`).

**Review by brb — CHANGES_REQUESTED** [review]

File: daemon/datapath.go (hunk 1 above)
Comment (borkmann) [reply daemon/datapath.go:202]: Hmm, could you point out where in master?
```
# git grep -n RemoveRules
daemon/daemon.go:690:   RemoveRules()
daemon/datapath.go:208: d.iptablesManager.RemoveRules()
pkg/datapath/iptables/iptables.go:348:// RemoveRules removes iptables rules installed by Cilium.
pkg/datapath/iptables/iptables.go:349:func (m *IptablesManager) RemoveRules() {
```

Comment (brb) [reply daemon/datapath.go:202]: Ouch, my bad - for some reasons I thought that we call from those places directly `RemoveRules()`, but instead `compileBase()` is being called. So, no need to change anything from your side.

**Review by brb — APPROVED** [review]

**Pull request merged**